### PR TITLE
Register all models from configuration as beans regardless of the existance of injection point

### PR DESCRIPTION
**Pull request by manovotn**
- Closes: #1059 

Just a draft to show how to enable injection even for models that do not have explicit injection points.

### CODE REVIEW

**Comment by geoand** [comment]
Thanks I'll have a look!

**Comment by geoand** [comment]
Ah, now I see what this does. I was under the impression that it was doing something more expansive, but what it does makes perfect sense

**Comment by geoand** [comment]
@manovotn whenever you think the PR is ready, please mark it as such

**Comment by manovotn** [comment]
I've expanded the tests and fix another bug that was causing `@Inject Instance<T>` to be treated as default model

File: core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/BeansProcessor.java [hunk 1]
```diff
@@ -298,6 +476,10 @@ private String determineModelName(InjectionPointInfo ip) {
                 return value;
             }
         }
+        // @Inject @Any Instance<Foo> should not be treated as default name
+        if (modelNameInstance == null && ip.isProgrammaticLookup()) {
```
Comment (geoand) [inline core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/BeansProcessor.java:480]: Nice, I was not aware of `isProgrammaticLookup`!

**Pull request merged**